projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/with-ai-extension.tsx

```diff
@@ -498,7 +504,7 @@ const blockEditWithAiComponents = createHigherOrderComponent( BlockEdit => {
 			<>
 				<BlockEdit { ...props } />
 
-				{ showAiControl && (
+				{ showAiControl && canExtendBlock && (
```

Comment:
The Input is not the only component, there is also the `AiAssistantExtensionToolbarDropdown`:

![2024-11-26_18-03](https://github.com/user-attachments/assets/324b586e-5a7e-406f-8071-14fa2f20e28f)

That said, I think the best way here may be to do an early return to bypass all those hook calls. When the module is disabled, there should be no performance penalty.

To avoid the React error with early returns, maybe we can do something like what is done [here](https://stackoverflow.com/a/56976469/4301486) with the `AfterEarlyReturn` component. 
If `isBlockExtensionEnabled` is false, return the original block, otherwise return the extended component. Makes sense?

Reply:
Good point! 
i did try making an early return, but that got me into this “page reload” scenario. I can show you, maybe you can spot something